### PR TITLE
[iceberg] Tag columns with "partition key" in DESCRIBE and SHOW COLUMNS output

### LINKED ISSUE #22638: Iceberg tables do not show partition key in DESCRIBE
## Your Environment

* Presto version used: latest master
* Storage (HDFS/S3/GCS..): S3
* Data source and connector used: Iceberg
* Deployment (Cloud or On-prem): On-prem

## Expected Behavior
Partition key has the value `partition key` in the `Extra` column of the DESCRIBE output like hive:
```
presto:tpcds_sf10000_parquet_varchar_iceberg_part> use hive.tpcds_sf10000_parquet_varchar_part;
USE
presto:tpcds_sf10000_parquet_varchar_part> describe store_sales;
        Column         |     Type     |     Extra     | Comment
-----------------------+--------------+---------------+---------
 ss_sold_time_sk       | integer      |               |
....omitted....
 ss_net_profit         | decimal(7,2) |               |
 ss_sold_date_sk       | bigint       | partition key |
(23 rows)
```
## Current Behavior
The partition key has an empty `Extra` value for partitioned Iceberg tables.
```
presto:tpcds_sf10000_parquet_varchar_iceberg_part> describe store_sales;;
        Column         |     Type     | Extra | Comment
-----------------------+--------------+-------+---------
 ss_sold_time_sk       | integer      |       |
 ss_item_sk            | integer      |       |
....omitted....
 ss_net_profit         | decimal(7,2) |       |
 ss_sold_date_sk       | bigint       |       |
(23 rows)

Query 20240430_054246_00031_fn54c, FINISHED, 3 nodes
Splits: 6 total, 6 done (100.00%)
[Latency: client-side: 379ms, server-side: 270ms] [23 rows, 2.59KB] [85 rows/s, 9.59KB/s]

presto:tpcds_sf10000_parquet_varchar_iceberg_part> show create table store_sales;
                                     Create Table
---------------------------------------------------------------------------------------
 CREATE TABLE iceberg.tpcds_sf10000_parquet_varchar_iceberg_part.store_sales (
    "ss_sold_time_sk" integer,
    "ss_item_sk" integer,
 ....omitted....
    "ss_net_profit" decimal(7,2),
    "ss_sold_date_sk" bigint
 )
 WITH (
    delete_mode = 'copy-on-write',
    format = 'PARQUET',
    format_version = '2',
    location = 's3a://presto-workload/tpcds-sf10000-parquet-iceberg-part/store_sales',
    partitioning = ARRAY['ss_sold_date_sk']
 )
(1 row)
```

### PR DESCRIPTION
**Pull request by imjalpreet**
## Description

After the change:
Screenshot 2024-05-06 at 2 02 50 PM

## Motivation and Context

Fixes #22638 

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by hantangwangd — COMMENTED** [review]
Thanks for the fix, a little question for discussing. Do you think we should also show the non-identity partition transform? For example when I create an Iceberg table as follows:

```
create table test_table(a int, b varchar, c timestamp) with (partitioning = ARRAY['a', 'truncate(b, 2)', 'year(c)']);
```

Then the `show create table` would show all the partition transforms:

```
presto:default> show create table test_table;
                             Create Table                              
-----------------------------------------------------------------------
 CREATE TABLE iceberg.default.test_table (                                    
    "a" integer,                                                       
    "b" varchar,                                                       
    "c" timestamp                                                      
 )                                                                     
 WITH (                                                                
    delete_mode = 'merge-on-read',                                     
    format = 'PARQUET',                                                
    format_version = '2',                                              
    location = 'file:/Users/wangd/work/data/iceberg/data/default/test_table', 
    partitioning = ARRAY['a','truncate(b, 2)','year(c)']               
 )          
```

Maybe it's better to keep `desc` be consistent with `show create table`, what's your opinion?

**Review by elharo — COMMENTED** [review]
needs tests

**Comment by imjalpreet** [comment]
>  Do you think we should also show the non-identity partition transform?

@hantangwangd I think that's a valid ask. What would you suggest we should mention in the Extra info in this case? Should we mention the transformation being used for the hidden partitioning for the respective columns?

**Comment by hantangwangd** [comment]
@imjalpreet For non-identity partition column, I think show the transform information would be enough, maybe something like `partition by '<transform.toString()>'`. So that it would be shown as follows:

```
presto:default> desc test_table;
 Column |   Type    |           Extra            | Comment 
--------+-----------+----------------------------+---------
 a      | integer   | partition key              |         
 b      | varchar   | partition by 'truncate[2]' |         
 c      | timestamp | partition by 'year'        |         
(3 rows)

presto:default> show columns in test_table;
 Column |   Type    |           Extra            | Comment 
--------+-----------+----------------------------+---------
 a      | integer   | partition key              |         
 b      | varchar   | partition by 'truncate[2]' |         
 c      | timestamp | partition by 'year'        |         
(3 rows)

```

Is that ok? Or do you have a better idea?

**Comment by imjalpreet** [comment]
@hantangwangd let's say for a date/timestamp column we have two hidden partition transforms year and month. What would be the best way to display that? Should we write partition by year, month or is there a better way to communicate that there are two hidden partition transforms on this column?

**Comment by imjalpreet** [comment]
@hantangwangd thanks for your inputs and sorry for the delay. I was finally able to resume on this PR and I have pushed the updates as we discussed. 

@elharo I have also added a couple of tests as requested.

**Review by hantangwangd — COMMENTED** [review]
@imjalpreet Thanks for your fix, overall looks good to me, some little things left.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java [hunk 1]
```diff
@@ -272,16 +273,47 @@ public static long getSnapshotIdAsOfTime(Table table, long millisUtc)
...
+        Map<String, List<String>> partitionFields = new HashMap<>();
+
+        switch (partitionType) {
+            case IDENTITY:
+                for (int i = 0; i < partitionSpec.fields().size(); i++) {
+                    PartitionField field = partitionSpec.fields().get(i);
+                    if (field.transform().isIdentity()) {
+                        List<String> transforms = new ArrayList<>();
+                        transforms.add(field.transform().dedupName());
+                        partitionFields.put(field.name(), transforms);
```
[the remark this answers is not in the document]
Comment (imjalpreet) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java]: Oh, thanks I missed that `dedupName` is not overridden for each sub-transform. I have updated it to use `toString`.

**Review by hantangwangd — APPROVED** [review]
LGTM, thanks for the fix.

**Pull request merged**